assets/src/console/pages/gateway-accounts/CardsPage.tsx

```diff
@@ -0,0 +1,120 @@
+import * as React from "react";
+import { Helmet } from "react-helmet";
+import createDecorator from "final-form-focus";
+
+import {
+  PageTitle,
+  P,
+  Checkbox,
+  Button,
+  Form,
+  Hr,
+  Strong,
+  Loader,
+  ErrorAlert
+} from "@pay/web";
+
+import {
+  GatewayAccountFragment,
+  ServiceFragment,
+  useSubmitCardTypeMutation
+} from "../../__generated__/graphql";
+import { useCardTypesQuery } from "../../../platform-admin/__generated__/graphql";
+
+interface FormValues {
+  cardTypes: Array<string>;
+}
+
+const decorators = [createDecorator<FormValues>()];
+
+interface Props {
+  service: ServiceFragment;
+  gatewayAccount: GatewayAccountFragment;
+}
+
+const CardsPage: React.FC<Props> = ({ service, gatewayAccount }) => {
+  const [submitCardType] = useSubmitCardTypeMutation();
+  const { loading, error, data } = useCardTypesQuery({ errorPolicy: "all" });
+
+  return (
+    <>
+      <Helmet>
+        <title>Manage payment types</title>
+      </Helmet>
+      <PageTitle title="Manage payment types" />
+      <P>Choose which credit and debit cards you want to accept</P>
+      <Form<FormValues> //TODO: Refactor this code with Fieldset
+        initialValues={{
+          cardTypes: gatewayAccount.cardTypes.map(card => card.id)
+        }}
+        onSubmit={async values => {
+          await submitCardType({
+            variables: {
+              gatewayAccountId: gatewayAccount.id,
+              cardTypeIds: values.cardTypes
+            }
+          });
+        }}
+        column
+        decorators={decorators}
+      >
+        <React.Fragment>
+          <Strong>Debit Cards</Strong>
+          {loading ? (
+            <Loader message="Loading payment card types" />
+          ) : error || !data ? (
+            <ErrorAlert
+              title="Unable to retrieve card types"
+              message={error && error.message}
+              showError
+            />
+          ) : (
+            data.cardTypes
+              .filter(card => card.type === "DEBIT")
+              .map(card => {
+                return (
+                  <>
+                    <Hr />
+                    <Checkbox
+                      name="cardTypes"
+                      label={
+                        (card.label === "Visa" || card.label === "Mastercard"
```

Comment:
not sure what the intent of this logic is here. it seems like it's best put in a function at the top of this file. also, you're using the label as part of the logic. is there any way to avoid using the label? also, you're using `!` a few times here which is a bit risky. if you refactor this into a utility function and just return early if things are not defined, that's probably safer.

Reply:
i think you can use the `brand` property not `label` (I think brand should be an enum)